armory/datasets/generator.py

```diff
@@ -0,0 +1,135 @@
+"""
+Example:
+
+from datasets import load, generator, preprocessing
+info, ds = load.load("digit")
+gen = generator.ArmoryDataGenerator(info, ds, element_map = lambda z: (preprocessing.audio_to_canon(z["audio"]), z["label"]))
+x, y = next(gen)
+
+info, ds = load.load("mnist")
+gen = generator.ArmoryDataGenerator(info, ds, element_map = lambda z: (preprocessing.image_to_canon(z["image"]), z["label"]), framework="tf", batch_size=5)
+x, y = next(gen)
+
+"""
+
+import tensorflow as tf
+import tensorflow_datasets as tfds
+
+
+# NOTE: currently does not extend art.data_generators.DataGenerator
+#     which is necessary for using ART `fit_generator` method
+class ArmoryDataGenerator:
+    """
+    Returns batches of numpy data
+
+    Raw = no preprocessing
+
+    ds_dict = dataset dictionary without split selected
+
+    num_batches - if not None, the number of batches to grab
+
+    element_filter - predicate of which elements to keep; occurs prior to mapping
+        Note: size computations will be wrong when filtering is applied
+    element_map - function that takes a dataset element (dict) and maps to new element
+    """
+
+    FRAMEWORKS = ("tf", "numpy", "torch")
+
+    def __init__(
+        self,
+        info,
+        ds_dict: dict,
```

Comment:
could be helpful to spell out in a docstring what they keys in `ds_dict` should map to, or how one constructs said dict

Reply:
Added in docstring.